CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+cmake_minimum_required (VERSION 3.8.2)
+project(imgui VERSION 1.73.0 LANGUAGES CXX C)
+
+option (IMGUI_EXAMPLES "Build ImGui examples" ON)
+option (IMGUI_DEMO "Include the ImGui demo window implementation in library" ON)
+
+option (IMGUI_IMPL_SDL "Build the SDL implementation (only if supported)" ON)
+option (IMGUI_IMPL_METAL "Build the Metal implementation (only if supported)" ${APPLE})
+option (IMGUI_IMPL_OSX "Build the OSX implementation (only if supported)" ${APPLE})
+option (IMGUI_IMPL_WIN32 "Build the Win32 (native winapi) implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_GLFW "Build the GLFW implementation (only if supported)" ON)
+option (IMGUI_IMPL_GLUT "Build the GLUT implementation (only if supported)" ON)
+option (IMGUI_IMPL_OPENGL "Build the OpenGL implementation (only if supported)" ON)
+option (IMGUI_IMPL_OPENGL2 "Build the OpenGL 2 (legacy) implementation (only if supported)" ${IMGUI_IMPL_OPENGL})
+option (IMGUI_IMPL_DX9 "Build the DirectX 9 implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_DX10 "Build the DirectX 10 implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_DX11 "Build the DirectX 11 implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_DX12 "Build the DirectX 12 implementation (only if supported)" ${WIN32})
+
+add_library (imgui
+    imgui.cpp
+    imgui_draw.cpp
+    imgui_widgets.cpp
+    $<$<BOOL:IMGUI_DEMO>:imgui_demo.cpp>
+)
+
+add_library(imgui::imgui ALIAS imgui)
+
+target_include_directories (imgui PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
```

Comment:
You better add variables for `add_subdirectory` and `install` usage.
```
target_include_directories(imgui PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include>)
```

Furthermore, it is a good practice to store headers in `include/lib_name` folder. By including the whole source dir you pollute includes. Using a separate `include/imgui` folder for headers will make it clear for library users to understand which library the included header belongs to:
`#include <imgui/imgui.h>`
https://cliutils.gitlab.io/modern-cmake/chapters/basics/structure.html - a good reference for modern cmake

Reply:
I'm aware of how good projects are structured. As I've stated elsewhere in this PR, this was modeled to match the existing build system and layout.